addon/mixins/new-popover-modal-api.js

```diff
@@ -0,0 +1,30 @@
+export default Ember.Mixin.create({
+  init(...args) {
+    this._super(...args);
+    let opts = this.get('_propertyOptions');
+    let isNewApi = this.get('__renderPopoverNewAPI');
+    Ember.assert(`[ember-widgets] popover or modal must be initialized with _propertyOptions`, !isNewApi || !!opts);
```

Comment:
Is there a higher level error that can be thrown here? "A component extending from ember-widget's model or popover base classes cannot be invoked directly. Use the popover service."

If `this.get('__renderPopoverNewAPI')` it so be stripped it need so be part of the `assert` line:

```js
Ember.assert(
  `Some error`,
  !(this.get('__renderPopoverNewAPI')) || !!opts
);
```

Reply:
Good call...made the change